pkg/cmd/pulumi/convert_test.go

```diff
@@ -32,18 +33,20 @@ func TestYamlConvert(t *testing.T) {
 		t.Fatalf("Pulumi.yaml is a directory, not a file")
 	}
 
-	result := runConvert("convert_testdata", []string{}, "yaml", "go", "convert_testdata/go", true)
+	result := runConvert(env.Global(), "convert_testdata", []string{}, "yaml", "go", "convert_testdata/go", true)
 	require.Nil(t, result, "convert failed: %v", result)
 }
 
-//nolint:paralleltest // sets env var, must be run in isolation
 func TestPclConvert(t *testing.T) {
-	t.Setenv("PULUMI_DEV", "TRUE")
+	t.Parallel()
+	env := env.NewEnv(env.MapStore{
+		env.Dev.Var().Name(): "true",
+	})
 
 	// Check that we can run convert from PCL to PCL
 	tmp := t.TempDir()
 
-	result := runConvert("pcl_convert_testdata", []string{}, "pcl", "pcl", tmp, true)
+	result := runConvert(env, "pcl_convert_testdata", []string{}, "pcl", "pcl", tmp, true)
```

Comment:
This is an example of injecting the environment into a function.

An environment is anything that can map a name to a typed value, and it can be easily generated from a `map`.

Reply:
Love it!